### PR TITLE
Init models

### PR DESCRIPTION
**Pull request by trantube94**
I have created a new pull request , check it please!

### CODE REVIEW

File: app/models/answer.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Answer < ActiveRecord::Base
+  belongs_to :answer, class_name: "Word"
```
Comment (phamkykhoi) [inline app/models/answer.rb]: Not use relation. Remove

File: app/models/answer.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  belongs_to :question, class_name: "Word"
```
Comment (phamkykhoi) [inline app/models/answer.rb]: Remove class_name: "Word"

File: app/models/category.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Category < ActiveRecord::Base
+  validates :name, presence: true
+  validates :total, presence: true
+  has_many :lessons
```
Comment (phamkykhoi) [inline app/models/category.rb]: dependent: :destroy

File: app/models/category.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+  has_many :words
```
Comment (phamkykhoi) [inline app/models/category.rb]: dependent: :destroy

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  validates :password, presence: true, length: { minimum: 6 }
+  has_many :active_relationships, class_name:  "Relationship",
+                                  foreign_key: "follower_id",
+                                  dependent:   :destroy
+  has_many :passive_relationships, class_name:  "Relationship",
+                                   foreign_key: "followed_id",
+                                   dependent:   :destroy
+  has_many :following, through: :active_relationships, source: :followed
+  has_many :followers, through: :passive_relationships, source: :follower
+  has_many :lessons
```
Comment (phamkykhoi) [inline app/models/user.rb]: dependent: :destroy

File: app/models/category.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Category < ActiveRecord::Base
+  validates :name, presence: true
+  validates :total, presence: true
+  has_many :lessons, dependent: :destroy
```
Comment (phamkykhoi) [inline app/models/category.rb]: Move to before validates

File: app/models/word.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class Word < ActiveRecord::Base
+  validates :content, presence: true
+  belongs_to :category
+  has_many :results, dependent: :destroy
```
Comment (phamkykhoi) [inline app/models/word.rb]: Move to before validates

File: config/database.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# SQLite version 3.x
```
Comment (dieunb) [inline config/database.yml]: don't push this file

File: app/models/answer.rb [hunk 9]
```diff
@@ -0,0 +1,4 @@
+class Answer < ActiveRecord::Base
+  belongs_to :answer
```
Comment (phamkykhoi) [inline app/models/answer.rb]: Sao lại belongs_to :answer

File: app/models/category.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+class Category < ActiveRecord::Base
+  has_many :lessons, dependent: :destroy
+  has_many :words, dependent: :destroy
+
+  validates :name, presence: true
+  validates :total, presence: true
```
Comment (phamkykhoi) [inline app/models/category.rb]: what is total ?

Comment (trantube94) [inline app/models/category.rb]: Total is the number questions in a category
Example: Basic500   - lesson has 20 questions 
                Basic1000 - lesson has 30 questions

Comment (phamkykhoi) [inline app/models/category.rb]: Cái này không cần em nhé. Em bỏ đi

File: app/models/user.rb [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+                                  dependent:   :destroy
+  has_many :passive_relationships, class_name:  "Relationship",
+                                   foreign_key: "followed_id",
+                                   dependent:   :destroy
+
+  has_many :following, through: :active_relationships, source: :followed
+  has_many :followers, through: :passive_relationships, source: :follower
+  has_many :lessons, dependent: :destroy
+
+  validates :name,  presence: true, length: {maximum: 50}
```
Comment (phamkykhoi) [inline app/models/user.rb]: Move 50 to Settings

File: app/models/user.rb [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
```
Comment (phamkykhoi) [inline app/models/user.rb]: Move 255 to Settings

File: app/models/user.rb [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+                    format: {with: VALID_EMAIL_REGEX},
+                    uniqueness: {case_sensitive: false}
+  has_secure_password
+  validates :password, presence: true, length: {minimum: 6}
```
Comment (phamkykhoi) [inline app/models/user.rb]: Move 6 to Settings

**Pull request merged**